pygfunction/gfunction.py

```diff
@@ -2344,11 +2344,13 @@ def _map_segment_pairs(self, i_pair, j_pair, k_pair, borehole_to_borehole,
             if type(self.nSegments) == int:
                 nq_i = self.nSegments
                 nq_j = self.nSegments
+                i_sum = i * nq_i
+                j_sum = j * nq_j
             else:
-                nq_i = self.nSegments[i]  # number of sources in borehole i
-                nq_j = self.nSegments[j]  # number of sources in borehole j
-            i_segment.append(i_pair + i * nq_i)
-            j_segment.append(j_pair + j * nq_j)
+                i_sum = sum(self.nSegments[0:i])
+                j_sum = sum(self.nSegments[0:j])
+            i_segment.append(i_pair + i_sum)
+            j_segment.append(j_pair + j_sum)
```

Comment:
With the example you give, I was getting an out of bounds error with the `h_ij` matrix. I had made a mistake, its now corrected.

Reply:
However, I am now getting a Singular matrix error. 

```
  File "/media/jackcook/SHARED/Masters/Research/pygfunction/pygfunction/gfunction.py", line 1303, in solve
    X = np.linalg.solve(A, B)
  File "<__array_function__ internals>", line 6, in solve
  File "/home/jackcook/anaconda3/envs/GLHE/lib/python3.6/site-packages/numpy/linalg/linalg.py", line 394, in solve
    r = gufunc(a, b, signature=signature, extobj=extobj)
  File "/home/jackcook/anaconda3/envs/GLHE/lib/python3.6/site-packages/numpy/linalg/linalg.py", line 88, in _raise_linalgerror_singular
    raise LinAlgError("Singular matrix")
numpy.linalg.LinAlgError: Singular matrix
```

Not sure what the problem is at the moment.

Reply:
There appears to be some same-segment to same-segment responses missing in `h_dt`.